### PR TITLE
Update for Swift3

### PR DESCRIPTION
**Pull request by zhigang1992**
hm, thought?

### CODE REVIEW

**Comment by mergesort** [comment]
@zhigang1992 Looks great. I made a few changes on top of your PR, maybe you want to take a look?

Things to note:
• I added `@discarableResult` so it would stop issuing warnings about not using the constraints.
• I added underscores so the API doesn't break for people upgrading their code bases.
• I added back the Objective-C API.

And a few other small things.

**Comment by zhigang1992** [comment]
Look great to me. 😄 

🍻

**Comment by zhigang1992** [comment]
You should tag current commit for 2.2 and 2.3 support. 😄

**Comment by mergesort** [comment]
Great idea. Will add those tags.

File: src/Anchorman.swift [hunk 1]
```diff
@@ -183,9 +184,10 @@ public extension UIView {
     }
 
     @discardableResult
-    func pinEdge(_ edge: EdgeAnchor, toEdge: EdgeAnchor, ofView: UIView, relation: NSLayoutRelation = .equal, constant: CGFloat = 0.0, priority: UILayoutPriority = UILayoutPriorityRequired, activate: Bool = true) -> NSLayoutConstraint {
-        let fromAnchor = self.layoutAnchor(edge: edge)
-        let toAnchor = ofView.layoutAnchor(edge: toEdge)
+    func pinEdge(_ edge: EdgeAnchor, toEdge: EdgeAnchor, ofView view: UIView, relation: NSLayoutRelation = .equal, constant: CGFloat = 0.0, priority: UILayoutPriority = UILayoutPriorityRequired, activate: Bool = true) -> NSLayoutConstraint {
+
+        let fromAnchor = edge.typedAnchorForView(view: self)
+        let toAnchor = edge.typedAnchorForView(view: view)
```
Comment (zhigang1992) [inline src/Anchorman.swift]: hah hah

**Comment by mergesort** [comment]
@zhigang1992 Can you open a separate PR for this? I can't do anything with it since the branch was merged already. Thanks!

**Comment by zhigang1992** [comment]
@mergesort already done. :D https://github.com/mergesort/Anchorman/pull/8

**Pull request merged**